src/parser/jobs/ast/modules/CelestialIntersection.tsx

```diff
@@ -0,0 +1,68 @@
+import {Trans} from '@lingui/react'
+import {ActionLink} from 'components/ui/DbLink'
+import ACTIONS from 'data/ACTIONS'
+import {CastEvent} from 'fflogs'
+import Module, {dependency} from 'parser/core/Module'
+import Suggestions, {SEVERITY, TieredSuggestion} from 'parser/core/modules/Suggestions'
+import React from 'react'
+
+const SEVERITY_MOD = {
+	MINOR: 0.15,
+	MEDIUM: 0.5,
+	MAJOR: 0.8,
+}
+
+// Lifted from WHM benison and adapted to AST and TSX
+export default class CelestialIntersection extends Module {
+	static handle = 'celestialintersection'
+
+	@dependency private suggestions!: Suggestions
+
+	private lastUse = 0
+	private uses = 0
+	private totalHeld = 0
+
+	protected init() {
+		this.addHook('cast', {abilityId: ACTIONS.CELESTIAL_INTERSECTION.id, by: 'player'}, this.onCast)
+		this.addHook('complete', this.onComplete)
+	}
+
+	private onCast(event: CastEvent) {
+		this.uses++
+		if (this.lastUse === 0) { this.lastUse = this.parser.fight.start_time }
+
+		const held = event.timestamp - this.lastUse - (ACTIONS.CELESTIAL_INTERSECTION.cooldown * 1000)
+		if (held > 0) {
+			this.totalHeld += held
+		}
+		// update the last use
+		this.lastUse = event.timestamp
+	}
+
+	onComplete() {
+		const holdDuration = this.uses === 0 ? this.parser.fightDuration : this.totalHeld
+		const usesMissed = Math.floor(holdDuration / (ACTIONS.CELESTIAL_INTERSECTION.cooldown * 1000))

```

Comment:
might be worth factoring the multiplied cd out somewhere, it's being used in a fair few places